### ISSUE TITLE
Ugly NullReferenceException if not in a git repo

### ISSUE THREAD

**Issue by roji** [GitTools/GitVersion#853]
The following seems a bit much (GitVersion 1.0.0.1):

```
C:\>gitversion
INFO [05/06/16 13:41:29:08] Working directory: C:\
ERROR [05/06/16 13:41:29:11] An unexpected error occurred:
System.NullReferenceException: Object reference not set to an instance of an object.
   at GitVersion.GitPreparer.GetDotGitDirectory()
   at GitVersion.ExecuteCore.ExecuteGitVersion(String targetUrl, String dynamicRepositoryLocation, Authentication authentication, String targetBranch, Boolean noFetch, String workingDirectory, String commitId)
   at GitVersion.SpecifiedArgumentRunner.Run(Arguments arguments, IFileSystem fileSystem)
   at GitVersion.Program.VerifyArgumentsAndRun()
INFO [05/06/16 13:41:29:11]
INFO [05/06/16 13:41:29:11] Here is the current git graph (please include in issue):
INFO [05/06/16 13:41:29:11] Showing max of 100 commits
INFO [05/06/16 13:41:29:17] ERROR: fatal: Not a git repository (or any of the parent directories): .git
```

**Comment by dazinator** [comment]
Yep I just reproduced this using the latest `GitVersion.Portable` chocolatey package. (GitVersion 3.5.2).

Ran into it because I was configuring Team City step, and hadn't switched to "Agent Checkout". Agree that the error message here should be nicer :)

**Comment by asbjornu** [comment]
Should be fixed in #860.
